src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2408,12 +2408,7 @@ public static void Reverse(ref byte buf, nuint length)
             }
 
             // Store any remaining values one-by-one
-            for (nuint i = 0; i < (length / 2); i++)
-            {
-                ref byte first = ref Unsafe.Add(ref buf, i);
-                ref byte last = ref Unsafe.Add(ref buf, length - 1 - i);
-                (last, first) = (first, last);
-            }
+            ReverseInner(ref buf, length);
```

Comment:
> Could you help elaborate why this is faster now?

It avoids this issue https://github.com/dotnet/runtime/pull/64412#discussion_r857798132 among other things.

Reply:
Note that this change also introduced 1.67x regression on arm64: https://github.com/dotnet/runtime/issues/68667. 

I agree that it would be nice for the JIT to optimize this to have the same performance as the original loop. None of our code-generators are there (each code-generator for different reasons).

Reply:
Can we get some numbers for this change on Arm64 to see if it also addresses the regression?

-- Just noting I'm not pushing back against this change, I think its fine and simplifies things. Just wanting to better understand why there is such a drastic difference here so we know what to look out for in the future.

Reply:
For reference, here is the code generated by the interpreter for the loop in question. The generated code is fairly similar with the arm64 code Jan posted above, with some additional redundancy in the unoptimized case. The main problem is that first and last are recomputed with every loop iteration, relative to `i`.

Current main:
```
IR_0007: conv.i8.i4     [48 <- 16],     // Computation of `first`, extra calculation due to non-constant indexer
IR_000a: mul.i8.imm     [48 <- 48], 4   //
IR_000e: add.i8         [24 <- 0 48],   /////////////////////
IR_0012: sub1.i4        [48 <- 8],      // Computation of `last`, extra calculation due to non-constant indexer
IR_0015: sub.i4         [48 <- 48 16],  //
IR_0019: conv.i8.i4     [48 <- 48],     //
IR_001c: mul.i8.imm     [48 <- 48], 4   //
IR_0020: add.i8         [48 <- 0 48],   ///////////////////
IR_0024: ldind.i4       [32 <- 24],
IR_0027: ldind.i4       [40 <- 48],
IR_002a: stind.i4       [nil <- 48 32],
IR_002d: stind.i4       [nil <- 24 40],
IR_0030: add1.i4        [16 <- 16],     // index must be incremented
IR_0033: ldc.i4.2       [48 <- nil],    // Division of length should be hoisted out of the loop, optimization not supported by interp yet
IR_0035: div.i4         [48 <- 8 48],   //
IR_0039: blt.i4.sp      [nil <- 16 48], IR_0007
```

This change:
```
IR_0015: ldind.i4       [32 <- 16],
IR_0018: ldind.i4       [40 <- 24],
IR_001b: stind.i4       [nil <- 16 40],
IR_001e: stind.i4       [nil <- 24 32],
IR_0021: add.i8.imm     [16 <- 16], 4
IR_0025: add.i8.imm     [24 <- 24], -4
IR_0029: clt.un.i8      [40 <- 16 24],
IR_002d: brtrue.i4.sp   [nil <- 40], IR_0015
```